hs_core/models.py

```diff
@@ -3348,9 +3356,12 @@ def size(self):
 
         Raises SessionException if iRODS fails.
         """
+        # trigger file size read for files that haven't been set yet
+        for f in self.files.filter(_size__lt=0):
+            f.size
         # compute the total file size for the resource
-        f_sizes = [f.size for f in self.files.all()]
-        return sum(f_sizes)
+        res_size = self.files.aggregate(Sum('_size'))
+        return res_size['_size__sum']
```

Comment:
@sblack-usu Thought ```res_size``` is a number, but it is actually a list?

Reply:
it's actually a dictionary.  Kind of weird huh?